nomad/plan_apply.go

```diff
@@ -149,6 +149,8 @@ func (s *Server) applyPlan(plan *structs.Plan, result *structs.PlanResult, snap
 	for _, alloc := range req.Alloc {
 		if alloc.CreateTime == 0 {
 			alloc.CreateTime = now
+		} else {
```

Comment:
I think this should always be updated (remove the else and always run). The reason I say this is that when the scheduler first creates the allocation we want `CreateTime == ModifyTime`